lib/spack/spack/test/detection.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import contextlib
+import glob
+import os
+import os.path
+
+import pytest
+
+import spack.config
+import spack.detection
+import spack.repo
+import spack.spec
+import spack.util.path
+import spack.util.spack_yaml
+
+
+@pytest.fixture
+def executables_found(monkeypatch):
+    def _factory(result):
+        def _mock_search(path_hints=None):
+            return result
+
+        monkeypatch.setattr(spack.detection.path, 'executables_in_path', _mock_search)
+    return _factory
+
+
+def test_find_external_single_package(mock_executable, executables_found):
+    pkgs_to_check = [spack.repo.get('cmake')]
+    executables_found({
+        mock_executable("cmake", output='echo "cmake version 1.foo"'): 'cmake'
```

Comment:
This PR might not be the place to fix this, but the `output` kwarg to `mock_executable` either needs to be changed to be `contents`, or the fixture needs to be changed to automatically prepend the `echo` portion.

Reply:
Outdated